### PR TITLE
Clarify distinction between PublicKeyCredentialUserEntity name and displayName

### LINKED ISSUE #1852: Emphasize use of `user.name` for RP's to help users distinguish credentials
## Context

RP's, like us at Cisco, use WebAuthn with a broad RP ID scoped in a way to allow for credential reuse across multiple "applications" on unique subdomains. Unfortunately, we've discovered that current, nascent passkeys management UIs emphasize displaying a credential's RP ID and username, which are often 

**Chrome:**
Screenshot 2023-02-13 at 6 14 57 PM

**1Password:**
Screenshot 2023-02-13 at 6 23 15 PM

**Safari (macOS):**
Screenshot 2023-02-13 at 6 16 38 PM

## Proposed Change

What if we added a "`note`" property somewhere in [PublicKeyCredentialCreationOptions](https://www.w3.org/TR/webauthn-2/#dictdef-publickeycredentialcreationoptions) to allow RP's to annotate a credential for the management interfaces to then expose to users for the user's ease of management?

Here are examples of places that "`note`" could go:

**1Password ("notes"):**
Screenshot 2023-02-13 at 6 15 44 PM

**Safari ("Notes"):**
Screenshot 2023-02-13 at 6 29 01 PM

Screenshot 2023-02-13 at 6 29 22 PM

This feels beneficial to both RP's and end users.

### PR DESCRIPTION
**Pull request by MasterKale**
This PR attempts to help RP's understand the differences between `user.name` and `user.displayName`. It most importantly establishes `user.name` as the primary value displayed by browsers to users. The definition of `user.displayName` has also been tweaked to emphasize that this value has utility to differentiate "sub-accounts" for a given `user.name`, but without encouraging its use as based on [observations I made back in February](https://github.com/w3c/webauthn/issues/1852#issuecomment-1429062522) only Android currently exposes this value.

Fixes #1852.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/pull/1932.html" title="Last updated on Sep 6, 2023, 5:42 PM UTC (aefe8f2)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webauthn/1932/5bd3dd1...aefe8f2.html" title="Last updated on Sep 6, 2023, 5:42 PM UTC (aefe8f2)">Diff</a>

### CODE REVIEW

File: index.bs
Comment (sbweeden) [inline index.bs:3262]: In prettty much every case I've ever seen, this gets set to the same value as user.name. Is it worth mentioning that is a perfectly acceptable thing to do?

**Review by emlun — APPROVED** [review]
LGTM, thanks!

File: index.bs
Comment (MasterKale) [reply index.bs]: @sbweeden I tried updating the text accordingly, let me know what you think. @emlun I'm curious to hear if the change I made today requires refinement, or if your current approval still stands.

**Review by nicksteele — APPROVED** [review]

File: index.bs
Comment (emlun) [inline index.bs:3265]: Continuing from https://github.com/w3c/webauthn/pull/1932#discussion_r1281092796 and [commit `f09e3d8c` comments](https://github.com/w3c/webauthn/commit/f09e3d8cea148765c5d634b283e9cc81f583ee6e#commitcomment-123354509):

Hm. Relaxing the last sentence from SHOULD to MAY is strictly speaking a normative change, though a backwards compatible one. It probably doesn't matter either way, but I don't think it needs to change either.

I think it reads marginally better with "The [=[RP]=]", if nothing else for consistency with the description of `name` in the previous section. I think "...when doing so" is a fine addition if the requirement is relaxed from SHOULD to MAY.

I'm not sure the idea of "sub-accounts" is necessary, but I wouldn't say it hurts either.

All in all I think this is still fine to merge as is, but since you asked @MasterKale :slightly_smiling_face: here's how I would phrase these changes as a whole:

```suggestion
    ::  A [=human palatability|human-palatable=] name for the [=user account=], intended only for display. For example, "Alex Müller" or "田中倫".
        [=Clients=] may display this value to help users differentiate between multiple [=credentials=] with the same value of {{PublicKeyCredentialEntity/name}}.

        The [=[RP]=] SHOULD let the user choose this, and SHOULD NOT restrict the choice more than necessary.
        The [=[RP]=] MAY set this to the same value as {{PublicKeyCredentialEntity/name}}.
```

Granted it might look a bit weird to have a SHOULD immediately followed by a conflicting MAY, but I think this should communicate that the latter is fine but the former is preferable when practical.

File: index.bs
Comment (emlun) [reply index.bs]: See reply in https://github.com/w3c/webauthn/pull/1932#discussion_r1281744220

File: index.bs
Comment (emlun) [reply index.bs:3265]: Again, to be clear: I think the PR is fine without the above suggestion, but feel free to use some, all or none of it as you like @MasterKale. :slightly_smiling_face:

Comment (MasterKale) [reply index.bs]: > ...but I think this should communicate that the latter is fine but the former is preferable when practical.

Ah, I see the subtle difference. Your suggested wording sounds more spec-like, and you captured the ultimate meaning I want to achieve with this effort. I made some changes accordingly.

Comment (emlun) [reply index.bs]: Ok, looks good!

**Review by emlun — APPROVED** [review]

File: index.bs
Comment (MasterKale) [reply index.bs]: @sbweeden I'd like to mark this resolve. Can you please review, and if everything looks good to you approve?

Comment (sbweeden) [reply index.bs]: lgtm - resolving

Comment (MasterKale) [reply index.bs]: Thanks for taking another look. Can you "Approve" the PR too when you get a chance?

**Review by akshayku — APPROVED** [review]

File: index.bs
Comment (MasterKale) [inline index.bs:3191]: This section needs to be tweaked to more strongly emphasize that this is **the** value for users to distinguish multiple credentials that may have the same email address (thinking about multi-tenant environments.)

I also want to add an example of something like "alex.mueller@example.com (production)" to open up perception to possible ways of differentiating multi-tenant credentials.

> [=Clients=] may display this value to help users differentiate between multiple
        [=credentials=] with the same value of {{PublicKeyCredentialEntity/name}}.

I should also remove this from the section **below**.

Comment (MasterKale) [reply index.bs:3191]: Example from @timcappalli:

```
dn: Tim Cappalli (Microsoft)
n: tim.cappalli@microsoft.com

is a common pattern I see
```

File: index.bs
Comment (akshayku) [inline index.bs:3261]: Remove this line because name is the unique identifier typically.

File: index.bs
Comment (MasterKale) [reply index.bs]: I took another stab at the wording here to try and emphasize that RP's should add to `name` to differentiate multiple credentials when otherwise all of them would end up with the same value.

File: index.bs
Comment (MasterKale) [reply index.bs]: I believe I've addressed this, if you can take another look @akshayku I'd appreciate it.

File: index.bs
Comment (emlun) [inline index.bs:3192]: Sorry, I think this still misses the mark a bit. One authenticator should almost never have two credentials for the same account, but may have credentials for different accounts.

Perhaps we're getting hung up on what an "account" is. You've mentioned that what you're trying to reflect here is a "multi-tenant" use case, where you have one user login account that contains multiple distinct domains of privileges, and uses different WebAuthn credentials to access each domain, but all under the same "umbrella" account. For example, a management console where one user manages several distinct application domains. I think the idea was that all of the user's domain-specific WebAuthn credentials would have the same `user.name`, for example the e-mail address used as a username to log in to the single "umbrella" account that manages all of those domains. Correct?

But within the context of WebAuthn, the term "[user account](https://w3c.github.io/webauthn/#user-account)" is defined as:

>a [user account](https://w3c.github.io/webauthn/#user-account) denotes the mapping of a set of [credentials](https://w3c.github.io/webappsec-credential-management/#concept-credential) [[CREDENTIAL-MANAGEMENT-1]](https://w3c.github.io/webauthn/#biblio-credential-management-1) to a (sub)set of a [Relying Party](https://w3c.github.io/webauthn/#relying-party)'s resources, as maintained and authorized by the [Relying Party](https://w3c.github.io/webauthn/#relying-party).

A WebAuthn credential doesn't need to be tied to an e-mail address, for example. It's perfectly fine for the "multi-tenant" application to associate a WebAuthn credential with a particular privilege domain instead of the "umbrella" account. The distinction here is that each of those privilege domains is a separate "user account" by the WebAuthn definition - they are not "sub-accounts" of the "umbrella" account, because the "umbrella" account is not a WebAuthn "user account" if no WebAuthn credentials are associated with it.

Does that make sense?

**Review by emlun — CHANGES_REQUESTED** [review]

File: index.bs
Comment (emlun) [reply index.bs:3192]: If so, perhaps it could suffice to just extend the examples?

```
          - When inherited by {{PublicKeyCredentialUserEntity}}, it is a
            [=human palatability|human-palatable=] identifier for a [=user account=]. This
            identifier is the primary value displayed to users by [=Clients=] to help users
            understand with which [=user account=] a credential is associated.

            Examples of suitable values for this identifier include, "alexm", "+14255551234",
            "alex.mueller@example.com", "alex.mueller@example.com (ACME Corporation)",
            or "alex.mueller@example.com (ОАО Примертех)".

```

File: index.bs
Comment (agl) [inline index.bs:3264]: "The [=[RP]=] MAY set this to the same value as {{PublicKeyCredentialEntity/name}}."

I think we should suggest setting the empty string when no suitable value is available, rather than duplicating.

Comment (MasterKale) [reply index.bs]: I've updated the wording here to suggest it's okay to set this to an empty string if needed.

File: index.bs
Comment (MasterKale) [reply index.bs]: > You've mentioned that what you're trying to reflect here is a "multi-tenant" use case, where you have one user login account that contains multiple distinct domains of privileges, and uses different WebAuthn credentials to access each domain, but all under the same "umbrella" account. For example, a management console where one user manages several distinct application domains.

This exactly. For most RP's `user.name` will be as simple as an email address or username. But for a cloud hosting PaaS, Alex could be the admin of multiple customers' VPS' so the PaaS RP should append some unique value per VPS to his PaaS username when they allow Alex to register authenticators for logging into those admin panels using WebAuthn. 

I've incorporated your suggestions and tweaked the examples accordingly. Let me know what you think.

File: index.bs
Comment (MasterKale) [reply index.bs]: @agl Can you please take a look at this latest change and see if it satisfies your concern?

File: index.bs
Comment (MasterKale) [reply index.bs]: @emlun Can you please take a look at this latest change and see if it satisfies your concern?

File: index.bs
Comment (emlun) [inline index.bs:3193]: I think with the addition of the (prod-env) example, we can drop the (ACME Corp.) one.

```suggestion
            "alex.mueller@example.com", "alex.mueller@example.com (prod-env)",
            or "alex.mueller@example.com (ОАО Примертех)".
```

File: index.bs
Comment (emlun) [inline index.bs:3266]: I think these examples should still include the one without (ACME) for contrast.

```suggestion
        Examples of suitable values for this identifier include, "Alex Müller", "Alex Müller (ACME Co.)" or "田中倫".
```

**Review by emlun — APPROVED** [review]
Looks good to me. :slightly_smiling_face: The attached suggestions are optional.

File: index.bs
Comment (MasterKale) [reply index.bs]: Changes made ✌️

File: index.bs
Comment (MasterKale) [reply index.bs]: Changes made ✌️

**Pull request merged**
